### DISCUSSION TITLE
how to add new user  delete user block user  unblock user change password

### DISCUSSION THREAD

**Discussion by pphee** [zitadel/zitadel#7633] [Q&A]
I am working on a user management system in Go and need some guidance on implementing several key features. Could you please share insights or examples on how to:

Add a new user
Delete a user
Block a user
Unblock a user
Change a user's password
I'm using Go (Golang) for this project

Thank you!

**Comment by hifabienne** [comment]
Hi @pphee 
I guess you want to implement that by using our APIs.
You can find all those endpoints in out API docs:
https://zitadel.com/docs/category/apis/resources/mgmt/users
https://zitadel.com/docs/category/apis/resources/mgmt/user-human

Some of the endpoints are specific to human users (e.g create, and change password), and some are generalized like delete, deactivate, reactivate, etc. so depending on that you can use the users or user human api enpoints

We also have a guide on how to setup everything for your backend app includind an application and a service user: 
https://zitadel.com/docs/examples/secure-api/go
This is all based on our go SKD you can find here: https://github.com/zitadel/zitadel-go

**Comment by pphee** [comment]
Hello, I am encountering an authentication issue while working with an RPC service and need help fixing it. The error message is as follows:

{
    "error": "rpc error: code = Unauthenticated desc = transport: per-RPC creds failed due to error: http status not ok: 400 Bad Request {\"error\":\"invalid_request\",\"error_description\":\"assertion invalid\"}\n"
}

It appears to be related to invalid credentials or request assertion. Has anyone faced this issue before, and how can I resolve it? Any advice or guidance would be greatly appreciated. Thank you!

**Reply by hifabienne** [reply]
Hei @pphee 
Without more information it is hard to tell where the problem could be.
I assume you are trying to authenticate a service user?
What kind of authentication did you use?
Which guide or documentation do you follow?
What endpoint do you call when you get that error?
What are you trying to do?

Here you can find the docs about authenticating service users and the different authentication methods you have and how to use them: 
https://zitadel.com/docs/guides/integrate/service-users/authenticate-service-users
